### DISCUSSION TITLE
Cannot parse a Markdown file that has `import.meta.env`

### DISCUSSION THREAD

**Discussion by NicChoi** [vitejs/vite-plugin-vue#180] [Q&A]
### Issue Description
I'm using 'vite-plugin-vue-markdown' to parse the Markdown file (Which has the contents '\`import.meta.env\`') as a Vue component.  
When I use 'yarn dev' to run the dev server, the project can run and the page can be displayed normally.  
But when I use 'yarn build' to build the project, Vite reports the Error:
```
yarn run v1.22.19
$ vite build
vite v4.2.1 building for production...
✓ 7 modules transformed.
✓ built in 264ms
[vite:build-import-analysis] Parse error @:1:1
file: /Users/cai/NicChoi/vite-md-bug/src/assets/demo2.md
error during build:
Error: Parse error @:1:1
    at parse$e (file:///Users/cai/NicChoi/vite-md-bug/node_modules/vite/dist/node/chunks/dep-79892de8.js:16018:355)
    at Object.transform (file:///Users/cai/NicChoi/vite-md-bug/node_modules/vite/dist/node/chunks/dep-79892de8.js:45421:27)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Reproduce
- Github Repo: https://github.com/NicChoi/vite-md-bug
- Steps
  - `yarn`
  - `yarn dev`
    - Open the link and the page will show
  - `yarn build`
    - Cannot build

**Comment by NicChoi** [comment]
Has anyone encountered the same issue as me?

**Comment by patak-cat** [comment]
This is expected, see https://github.com/vitejs/vite/discussions/13387#discussioncomment-6049612
